### PR TITLE
Documentation for Cache rework

### PR DESCRIPTION
**Pull request by hermitdave**
Updated docs for Cache rework

### CODE REVIEW

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+---
+permalink: /en-US/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.htm
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md]: Why has this file been touched/added? It doesn't seem related to your PR.

Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md]: Never mind, I can see your other PR in the toolkit. Try and keep things a bit separate, it helps to establish what's going where.

Comment (hermitdave) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md]: sorry chap.. thought I do both at the same time..

File: en-us/api/Microsoft_Toolkit_Uwp_UI.md [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
+
+
+| Class | Description |
+| --- | --- |
+| [BoolToVisibilityConverter](Microsoft_Toolkit_Uwp_UI_Converters_BoolToVisibilityConverter.htm) | This class converts a boolean value into a Visibility enumeration. |
+| [CollectionVisibilityConverter](Microsoft_Toolkit_Uwp_UI_Converters_CollectionVisibilityConverter.htm) | This class converts a collection size to visibility. |
+| [ConverterTools](Microsoft_Toolkit_Uwp_UI_Converters_ConverterTools.htm) | Static class used to provide internal tools |
+| [StringFormatConverter](Microsoft_Toolkit_Uwp_UI_Converters_StringFormatConverter.htm) | This class provides a binding converter to display formatted strings |
+| [StringVisibilityConverter](Microsoft_Toolkit_Uwp_UI_Converters_StringVisibilityConverter.htm) | This class converts a string value into a Visibility value (if the value is null or empty returns a collapsed value). |
+| [InMemoryStorage](Microsoft_Toolkit_Uwp_UI_InMemoryStorage.htm) | Provides methods to store instances of any type in memory |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI.md]: Everything above this line is already in the md file, why are they showing as new?

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+## Members
+
+The **CacheBase<T>** class has the following types of members:
+
+### Methods
+
+#### InitializeAsync(Windows.Storage.StorageFolder folder, System.String folderName)
+
+this method is used supply a folder within which cache specific folder is created based on name provided.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: Please start sentences with a capital letter

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| folder | folder where cache specific folder is created | [StorageFolder](https://msdn.microsoft.com/en-us/library/windows/apps/windows.storage.storagefolder.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: Same goes for descriptions

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+## Members
+
+The **FileCache** class has the following types of members:
+
+### Methods
+
+#### InitializeAsync(Windows.Storage.StorageFolder folder, System.String folderName)
+
+this method is used supply a folder within which cache specific folder is created based on name provided.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: Please start sentences with a capital letter.

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| folder | folder where cache specific folder is created | [StorageFolder](https://msdn.microsoft.com/en-us/library/windows/apps/windows.storage.storagefolder.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: Same for descriptions.

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md [hunk 7]
```diff
@@ -1,60 +1,101 @@
...
+
+## Members
+
+The **ImageCache** class has the following types of members:
+
+### Methods
+
+#### InitializeAsync(Windows.Storage.StorageFolder folder, System.String folderName)
+
+this method is used supply a folder within which cache specific folder is created based on name provided.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md]: Please start sentences with a capital letter

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md [hunk 8]
```diff
@@ -1,60 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| folder | folder where cache specific folder is created | [StorageFolder](https://msdn.microsoft.com/en-us/library/windows/apps/windows.storage.storagefolder.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md]: Same goes for descriptions

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md [hunk 9]
```diff
@@ -1,60 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| returns | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### ClearAsync(System.TimeSpan duration)
+
+this method delete expired images from the cache and from within the in-memory cache.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md]: deletes, not delete

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md [hunk 10]
```diff
@@ -1,60 +1,101 @@
...
+
+##### Parameters
+###### | Name | Description | Type |
+
+| duration | this parameter is used to determine where image within cache has expired and should be delete or not. | [TimeSpan](https://msdn.microsoft.com/en-us/library/windows/apps/system.timespan.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md]: "determine whether the image within the cache and should be deleted or not"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md [hunk 11]
```diff
@@ -1,60 +1,101 @@
...
+
+| return | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### PreCacheAsync(System.Uri uri, System.String fileName, System.Boolean throwOnErro, System.Boolean storeToMemoryCache)
+
+This method downloads the image at uri supplied and saves it to the cache with the file name supplied. It also optionally loads the item into in-memory cache
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md]: "image at the uri supplied"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+description: API page for Microsoft.Toolkit.Uwp.UI.CacheBase
+keywords: windows, app, toolkit, UWP, API
+layout: api
+search.product: eADQiWindows 10XVcnh
+---
+
+
+# CacheBase<T> class
+
+Represents a strongly typed cache of objects that can be added to and removed as desired.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "added to and removed from." Remove "as desired."

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+The cache also provides in-memory storage of strongly typed objects to reduce reading from disc. The size of this in-memory storage is controlled by setting the ***MaxMemoryCacheCount*** property.
+The in-memory storage follows FIFO (first in first out) and the caller can pass optional parameter when calling ***PreCacheAsync*** to inform cache to use in-memory storage.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "can pass an optional parameter"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md (hunk 3 above)
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "This method is used to supply a StorageFolder and a cache folder name where the cache items will be stored"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md (hunk 4 above)
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "Folder where the cache"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+#### InitializeAsync(Windows.Storage.StorageFolder folder, System.String folderName)
+
+this method is used supply a folder within which cache specific folder is created based on name provided.
+
+##### Parameters
+######| Name | Description | Type |
+
+| folder | folder where cache specific folder is created | [StorageFolder](https://msdn.microsoft.com/en-us/library/windows/apps/windows.storage.storagefolder.aspx) |
+
+| folderName | the name given to cache specific folder | [String](https://msdn.microsoft.com/en-us/library/windows/apps/system.string.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "The name given to the cache specific folder"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| returns | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### ClearAsync(System.TimeSpan duration)
+
+this method delete expired items from the cache and from within the in-memory cache.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "deletes"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+###### | Name | Description | Type |
+
+| duration | this parameter is used to determine where item within cache has expired and should be delete or not. | [TimeSpan](https://msdn.microsoft.com/en-us/library/windows/apps/system.timespan.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "determine whether the item within the cache"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+
+| return | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### PreCacheAsync(System.Uri uri, System.String fileName, System.Boolean throwOnErro, System.Boolean storeToMemoryCache)
+
+This method downloads the file at uri supplied and saves it to the cache with the file name supplied. It also optionally loads the item into in-memory cache
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "at the uri supplied"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| uri | Uri of the item to precache. | [Uri](https://msdn.microsoft.com/library/windows/apps/System.Uri) |
+
+| fileName | the name used by cache when saving the item downloaded | [String](https://msdn.microsoft.com/en-us/library/windows/apps/system.string.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "by the cache"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+| throwOnError | optional parameter that allows for errors to be silently handled | [Boolean](https://msdn.microsoft.com/en-us/library/windows/apps/system.boolean.aspx) |
+
+| storeToMemoryCache | optional parameter that determines whether the item is also loaded to in-memory cache or not. | [Boolean](https://msdn.microsoft.com/en-us/library/windows/apps/system.boolean.aspx) |
+
+| return | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### GetFromCacheAsync(System.Uri uri, System.String fileName, System.Boolean throwOnError)
+
+Retrieves a strongly typed item from cache if it exists and has not expired. Otherwise the file is downloaded from uri supplied and item is returned.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "and the item is returned"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| uri | Uri of the item to download. | [Uri](https://msdn.microsoft.com/library/windows/apps/System.Uri) |
+
+| fileName | the name used by cache when saving the item downloaded | [String](https://msdn.microsoft.com/en-us/library/windows/apps/system.string.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "used by the cache when saving the downloaded item"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+
+| throwOnError | optional parameter that allows for errors to be silently handled | [Boolean](https://msdn.microsoft.com/en-us/library/windows/apps/system.boolean.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_CacheBase.md]: "Optional parameter that allows for errors to be thrown. Default is to silently handle them"
Reason I've changed this is because the default is already for it to be silently handled. Also, I think the description doesn't really match the name of the parameter.

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+description: API page for Microsoft.Toolkit.Uwp.UI.FileCache
+keywords: windows, app, toolkit, UWP, API
+layout: api
+search.product: eADQiWindows 10XVcnh
+---
+
+
+# FileCache class
+
+Represents a cache of StorageFile objects that can be added to and removed as desired.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "added to and removed from." Remove "as desired."

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md (hunk 5 above)
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "This method is used to supply a StorageFolder and a cache folder name where the cache items will be stored"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md (hunk 6 above)
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "where the cache"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+#### InitializeAsync(Windows.Storage.StorageFolder folder, System.String folderName)
+
+this method is used supply a folder within which cache specific folder is created based on name provided.
+
+##### Parameters
+######| Name | Description | Type |
+
+| folder | folder where cache specific folder is created | [StorageFolder](https://msdn.microsoft.com/en-us/library/windows/apps/windows.storage.storagefolder.aspx) |
+
+| folderName | the name given to cache specific folder | [String](https://msdn.microsoft.com/en-us/library/windows/apps/system.string.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "to the cache"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 24]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| returns | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### ClearAsync(System.TimeSpan duration)
+
+this method delete expired files from the cache and from within the in-memory cache.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "deletes"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 25]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+###### | Name | Description | Type |
+
+| duration | this parameter is used to determine where file within cache has expired and should be delete or not. | [TimeSpan](https://msdn.microsoft.com/en-us/library/windows/apps/system.timespan.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "determine whether the file within the cache has expired and should be deleted or not"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 26]
```diff
@@ -0,0 +1,101 @@
...
+
+| return | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### PreCacheAsync(System.Uri uri, System.String fileName, System.Boolean throwOnErro, System.Boolean storeToMemoryCache)
+
+This method downloads the file at uri supplied and saves it to the cache with the file name supplied. It also optionally loads the item into in-memory cache
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "at the uri supplied"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 27]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| uri | Uri of the file to precache. | [Uri](https://msdn.microsoft.com/library/windows/apps/System.Uri) |
+
+| fileName | the name used by cache when saving the file downloaded | [String](https://msdn.microsoft.com/en-us/library/windows/apps/system.string.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "used by the cache when saving the downloaded file"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 28]
```diff
@@ -0,0 +1,101 @@
...
+
+| throwOnError | optional parameter that allows for errors to be silently handled | [Boolean](https://msdn.microsoft.com/en-us/library/windows/apps/system.boolean.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: See remark in previous file about description for this parameter.

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 29]
```diff
@@ -0,0 +1,101 @@
...
+
+| storeToMemoryCache | optional parameter that determines whether the file is also loaded to in-memory cache or not. | [Boolean](https://msdn.microsoft.com/en-us/library/windows/apps/system.boolean.aspx) |
+
+| return | [Task](https://msdn.microsoft.com/en-us/library/windows/apps/system.threading.tasks.task.aspx) |
+
+
+#### GetFromCacheAsync(System.Uri uri, System.String fileName, System.Boolean throwOnError)
+
+Retrieves a file from cache if it exists and has not expired. Otherwise the file is downloaded from uri supplied and is returned.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "from the cache" "downloaded from the uri supplied"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 30]
```diff
@@ -0,0 +1,101 @@
...
+
+##### Parameters
+######| Name | Description | Type |
+
+| uri | Uri of the file to download. | [Uri](https://msdn.microsoft.com/library/windows/apps/System.Uri) |
+
+| fileName | the name used by cache when saving the file downloaded | [String](https://msdn.microsoft.com/en-us/library/windows/apps/system.string.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: "by the cache when saving the downloaded file"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md [hunk 31]
```diff
@@ -0,0 +1,101 @@
...
+
+| throwOnError | optional parameter that allows for errors to be silently handled | [Boolean](https://msdn.microsoft.com/en-us/library/windows/apps/system.boolean.aspx) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_FileCache.md]: Ditto here for previous remark

File: en-us/api/Microsoft_Toolkit_Uwp_UI_HyperlinkExtensions.md [hunk 32]
```diff
@@ -0,0 +1,29 @@
...
+layout: api
+search.product: eADQiWindows 10XVcnh
+---
+
+
+# HyperlinkExtensions class
+
+[Hyperlink](https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.xaml.documents.hyperlink.aspx) instances are used with Textblock or RichTextblock.
+At present there is no way to bind Hyperlink to an ICommand defined in ViewModel. 
+HyperlinkExtensions provides extension method that allows attaching command that is executed when [Click](https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.xaml.documents.hyperlink.click.aspx) event occurs.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_HyperlinkExtensions.md]: "provides an attached property that allows the attaching of a Command that is executed when the"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_HyperlinkExtensions.md [hunk 33]
```diff
@@ -0,0 +1,29 @@
...
+
+## Members
+
+The **HyperlinkExtensions** class has the following types of members:
+
+### Properties
+
+#### Command
+
+Gets or sets ICommand instance. The command is executed when hyperlink is clicked.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_HyperlinkExtensions.md]: "when the hyperlink"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md [hunk 34]
```diff
@@ -1,60 +1,101 @@
...
-
-
-### Properties
-
-#### CacheDuration
-
-Gets or sets the life duration of every cache entry.
-
-
+---
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ImageCache.md]: Rather that repeat myself again for this file, see the previous two cache files as nearly all will apply to this file too.

File: en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+description: API page for Microsoft.Toolkit.Uwp.UI.InMemoryStorage
+keywords: windows, app, toolkit, UWP, API
+layout: api
+search.product: eADQiWindows 10XVcnh
+---
+
+
+# InMemoryStorage<T> class
+
+Represents a strongly typed in-memory storage of objects that can be added / removed as required
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md]: Please keep some consistency with the description here, ideally use the same as previous cache docs

File: en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md [hunk 36]
```diff
@@ -0,0 +1,64 @@
...
+
+## Members
+
+The **InMemoryStorage<T>** class has the following types of members:
+
+### Methods
+
+#### ClearAsync()
+
+call this method to clear all items from in-memory storage.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md]: Please start sentences with a capital letter

File: en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md [hunk 37]
```diff
@@ -0,0 +1,64 @@
...
+
+
+#### ClearAsync(System.TimeSpan duration)
+
+call this method to clear items that are older than duration specified.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md]: "than the duration"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md [hunk 38]
```diff
@@ -0,0 +1,64 @@
...
+
+
+#### GetItem(System.String id, System.TimeSpan duration)
+
+Gets the item stored in-memory provided it has not expired. Expired item is removed and not returned to caller.
+
+##### Parameters
+######| Name | Description | Type |
+
+| id | key identifying the item. | [String](https://msdn.microsoft.com/library/windows/apps/System.String) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md]: Descriptions should start with a capital letter

File: en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md [hunk 39]
```diff
@@ -0,0 +1,64 @@
...
+
+| duration | Duration to check if item has expired or not. | [TimeSpan](https://msdn.microsoft.com/library/windows/apps/System.TimeSpan) |
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorage.md]: "if the item"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorageItem.md [hunk 40]
```diff
@@ -0,0 +1,38 @@
...
+description: API page for Microsoft.Toolkit.Uwp.UI.InMemoryStorageItem
+keywords: windows, app, toolkit, UWP, API
+layout: api
+search.product: eADQiWindows 10XVcnh
+---
+
+
+# InMemoryStorageItem<T> class
+
+Represents a strongly typed object that stored in InMemoryStorage<T>
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_InMemoryStorageItem.md]: "that is stored in"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md [hunk 41]
```diff
@@ -0,0 +1,25 @@
...
+description: API page for Microsoft.Toolkit.Uwp.UI.ListViewBaseExtensions
+keywords: windows, app, toolkit, UWP, API
+layout: api
+search.product: eADQiWindows 10XVcnh
+---
+
+
+# ListViewBaseExtensions class
+
+At present there is no way to use [ItemClick](https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.xaml.controls.listviewbase.itemclick.aspx) event of ListViewBase with MVVM.
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md]: "to use the"

File: en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md [hunk 42]
```diff
@@ -0,0 +1,25 @@
...
+ListViewBaseExtensions provides extension method that allow attaching ICommand to handle ListViewBase Item interaction using ICommand when ListViewBase [IsItemClickEnabled](https://msdn.microsoft.com/en-us/library/windows/apps/windows.ui.xaml.controls.listviewbase.isitemclickenabled.aspx) is set to true
```
Comment (ScottIsAFool) [inline en-us/api/Microsoft_Toolkit_Uwp_UI_ListViewBaseExtensions.md]: "provides an attached property that allows the attaching of an ICommand to"
